[EN/VI] Fixing typos, Updating Performance Troubleshooting and its VI translation (#5217)
ppy/osu-wiki: wiki/Performance_Troubleshooting/en.md

### Optimal

> This frame limiter attempts to keep a frame rate 8x your refresh rate, with a maximum of 960fps. This is our recommended frame limiter for users who don’t want to run their CPU/GPU to extremities, while maintaining high levels of performance resulting in low input latency.

Reviewer: ```suggestion
This frame limiter attempts to keep a frame rate 8x your refresh rate, with a maximum of 960 FPS. This is our recommended frame limiter for users who don’t want to run their CPU/GPU to extremities, while maintaining high levels of performance resulting in low input latency.
```

- GPUs generally aren’t meant to render 1000+ frames per second, so unexpected behaviour may occur, including lower performance or higher latency than otherwise attainable.

Even so, if heat generation and potentially overworking your hardware is not considered an issue, unlimited may be the option that provides you the most stable frame times.

> ### Custom

Reviewer 2: Is there any logic behind adding this to the article? It seems unnecessary and the ability to switch to this frame limiter was removed from in game for a reason. There are very few situations where changing this could ever help performance.

It's also very easy to do incorrectly, regardless of how detailed the instructions are.

Author: It has never been removed, it's never in the UI in the first place AFAIK.

Since fallback, there're 5 options: VSync, 120fps, 240fps, Unlimited (gameplay) and Unlimited.

And in latest (before the "Power Saving/Optimal" around last year), there were 4 options, same as above, except the Unlimited.

Although removed, you can still set to `FrameSync = CompletelyUnlimited` at that time, but well, no point in making the song select loads at max FPS.

But it was always 120/240 regardless of your monitor. And the third option, aka 240fps, can be set to a custom number via `FrameSync = 500fps` and `CustomFrameLimit = 500`. Only then, the game options would be: VSync, 120fps, 500fps, Unlimited (gameplay)

Author: So the point to set a custom frame limit can vary:
- I want my capped FPS to be a integer multiply of my monitor refresh rate. (let's say 2x. 4x, 8x, but it already got covered since Power Saving/Optimal update)
> GPUs generally aren’t meant to render 1000+ frames per second
- I have 60hz monitor and I want to set at 960, or 1000fps, but Optimal only provides me 480, and Unlimited burns my PC.
- I have a laptop and want to conserve battery, I don't want to consume much power with Optimal or Unlimited, but Power Saving gives me so much latency (because it's not osu!lazer that is multi-threadded), so I want a decent cap, like 4x refresh rate (instead of 2x and 8x marks).
- ...etc

Reviewer 2: Before the settings were changed to optimal and power saving, I believe the 240fps option would just change to whatever you had the custom FPS limit set to. So you were able to switch back and forth between your custom FPS and the other options.

I still think this is unnecessary to leave in, but if you are dead set on getting it added for whatever reason, you need to make sure the steps to changing it are correct and will actually work 100% of the time. Changing the cfg file without closing the game first will not work.

Author: > the 240fps option would just change to whatever you had the custom FPS limit set to as you mentioned. So you were able to switch back and forth between your custom FPS and the other options
It only changes the 240 to another number, not creating a new one though. And you would still need to change that manually in the cfg file.
> Changing the cfg file without closing the game first will not work.
Yes, this is true, I'll edit it.
> Opening the file in notepad before closing the game and then changing it may also not work.
You can open the file in the notepad normally, just don't save the file before closing the game.
We just need the value to be there and saved before opening the game.

- GPUs generally aren’t meant to render 1000+ frames per second, so unexpected behaviour may occur, including lower performance or higher latency than otherwise attainable.

Even so, if heat generation and potentially overworking your hardware is not considered an issue, unlimited may be the option that provides you the most stable frame times.

### Custom

> *Note that, if you're using the Cutting Edge release stream, changing the `CustomFrameLimit` to any number above 999, will be reverted to 999; on Stable, you can set the value above 1000 with no issues.*
> *Also note that you cannot switch into this setting with the `F7` shortcut in-game, even if you had done the steps above: you'll need to switch to Custom in the Settings before quitting the game so it'll be available the next time osu! runs. Otherwise, you'll have to change the `FrameSync` manually again.*

Reviewer: ```suggestion
*Notice to Cutting Edge users: changing the `CustomFrameLimit` to any number above 999, will be reverted to 999; on Stable, you can set the value above 1000 with no issues.*

*Caution: you cannot switch into this setting with the `F7` shortcut in-game, even if you do the steps below: you'll need to switch to Custom in the Settings before quitting the game so it'll be available the next time osu! runs. Otherwise, you'll have to change the `FrameSync` manually again.*
```

[hatnotes and notices have specific formatting](https://osu.ppy.sh/wiki/en/Article_styling_criteria/Formatting#notice)

also can't have two lines back-to-back because remark and also it's discouraged to have two notices back to back if im not mistaken. But I think it may be fine in this case because the article is so short

Author: I've changed it according to the suggestion, with ``changing the `CustomFrameLimit` to`` becomes ``changing the value of `CustomFrameLimit` to``.

This frame limiter attempts to keep a frame rate equal to your custom preference; however, this is a setting that cannot be changed to within the in-game menu by default.

To change osu!'s custom maximum framerate, you'll need to tweak some values in the config file:

1. Open your osu! folder by clicking the `Open osu! folder` button in the Options menu, or by navigating to the folder manually. (The default path is `C:\Users\<yourPCusername>\AppData\Local\osu!` for Windows; `/Applications/osu!.app/Contents/Resources/drive_c/osu!` for Mac.)
2. Close osu!.
3. Open the `osu!.<yourPCusername>.cfg` file with Notepad or a text editor of your choice.
4. Find the line `FrameSync = <value>` and change the value to `Custom`.
5. Find the line `CustomFrameLimit = <value>` and change the value to your preference.

> 6. Save the config file. If you haven't closed osu! in the second step, please do so before saving.

Reviewer: ```suggestion
6. Save the config file. (If you haven't closed osu! in the second step, please do so before saving.)
```

- GPUs generally aren’t meant to render 1000+ frames per second, so unexpected behaviour may occur, including lower performance or higher latency than otherwise attainable.

Even so, if heat generation and potentially overworking your hardware is not considered an issue, unlimited may be the option that provides you the most stable frame times.

### Custom

*Notice to Cutting Edge users: changing the value of `CustomFrameLimit` to any number above 999, will be reverted to 999; on Stable, you can set the value above 1000 with no issues.*

> *Caution: you cannot switch into this setting with the `F7` shortcut in-game, even if you do the steps below: you'll need to switch to Custom in the Settings before quitting the game so it'll be available the next time osu! runs. Otherwise, you'll have to change the `FrameSync` manually again.*

Reviewer 3: ```suggestion
*Caution: you cannot switch into this setting with the `F7` shortcut in-game, even if you do the steps below: you'll need to switch to `Custom` in the Settings before quitting the game so it'll be available the next time osu! runs. Otherwise, you'll have to change the `FrameSync` manually again.*
```

This frame limiter attempts to keep a frame rate equal to your custom preference; however, this is a setting that cannot be changed to within the in-game menu by default.

To change osu!'s custom maximum framerate, you'll need to tweak some values in the config file:

> 1. Open your osu! folder. (The default is `C:\Users\<yourPCusername>\AppData\Local\osu!` for Windows; `/Applications/osu!.app/Contents/Resources/drive_c/osu!` for Mac.)
> You can also click the "Open osu! folder" button in Options menu in-game.
> 2. Close osu!.
> 3. Open the `osu!.<yourPCusername>.cfg` file with Notepad or a text editor of your choice.
> 4. Find the line `FrameSync = <value>` and change the value to `Custom`.
> 5. Find the line `CustomFrameLimit = <value>` and change the value to your preference.
> 6. Save the config file. If you haven't close osu! in the second step, please do so before saving.
> 7. Re-open the game.

Reviewer: ```suggestion
1. Open your osu! folder. (The default is `C:\Users\<yourPCusername>\AppData\Local\osu!` for Windows; `/Applications/osu!.app/Contents/Resources/drive_c/osu!` for Mac.) 
  - Alternatively, you can open click the `Open current skin folder` button in Options menu and navigate up two folders.
2. Close osu!.
3. Open the `osu!.<yourPCusername>.cfg` file with Notepad or a text editor of your choice.
4. Find the line `FrameSync = <value>` and change the value to `Custom`.
5. Find the line `CustomFrameLimit = <value>` and change the value to your preference.
6. Save the config file. If you haven't close osu! in the second step, please do so before saving.
7. Re-open the game.
```

Reviewer 4: I'm very confused at this. Aside from grammatical issues, why click the "Open current skin folder" button when the "Open osu! folder" button does exactly what you'd want?

Reviewer: How did I never realize that button was there? I always just used the open current skin folder button lol

Author: I usually just right click osu! shortcut on desktop and select Open file location, lol.

Author: Thank you guys, I had changed to your 2nd suggestion.

- GPUs generally aren’t meant to render 1000+ frames per second, so unexpected behaviour may occur, including lower performance or higher latency than otherwise attainable.

Even so, if heat generation and potentially overworking your hardware is not considered an issue, unlimited may be the option that provides you the most stable frame times.

### Custom

> This frame limiter attempts to keep a frame rate equal to your custom preference. However, this is a setting that cannot be changed to within the in-game menu.

Reviewer: ```suggestion
This frame limiter attempts to keep a frame rate equal to your custom preference; however, this is a setting that cannot be changed to within the in-game menu by default
```

> To change osu! custom maximum framerate, you'll need to tweak some values in the config file. Here's the steps:

Reviewer: ```suggestion
To change osu!'s custom maximum framerate, you'll need to tweak some values in the config file:
```
colon can't be used after incomplete clause

Author: Thank you for your review. In my language, colon can be used in the middle of the sentence when splitting into different routes to connect with the incomplete clause.
I'll fix these.

> - Open your osu! folder. (The default is `C:\Users\<yourPCusername>\AppData\Roaming\osu!`.)
> - Open the `osu!.<yourPCusername>.cfg` file with Notepad or any text editors.
> - Find the line `FrameSync = <value>` and change the value to `Custom`.
> - Find the line `CustomFrameLimit = <value>` and change the value to your preference.

Reviewer: ```suggestion
1. Open your osu! folder. (The default is `C:\Users\<yourPCusername>\AppData\Roaming\osu!` for Windows; `/Applications/osu!.app/Contents/Resources/drive_c/Program Files/osu!/` for Mac.)
2. Open the `osu!.<yourPCusername>.cfg` file with Notepad or a text editor of your choice.
3. Find the line `FrameSync = <value>` and change the value to `Custom`.
4. Find the line `CustomFrameLimit = <value>` and change the value to your preference.
```
not sure about the full path for mac but needs to be there regardless

Reviewer 4: wouldn't it be quicker to refer to the `Open osu! folder`-button in the options?

Reviewer: That's also an option

Author: I'll check the wiki for the correct path for Mac: `/Applications/osu!.app/Contents/Resources/drive_c/osu!`
I've also change `Roaming` to `Local` in Windows path.
And thanks for your suggestion about the `Open osu! folder` in-game.

This is our recommended frame limiter for users who:

- Want to set the frame limiter at 4x refresh rate, etc. but the game doesn't have the option to do so.

> - Doesn't satisfied with the maximum of 960fps that Optimal gives (but **do not** want to run their CPU/GPU to extremities), resulting in even lower input latency than Optimal without abusing your CPU/GPU too much with Unlimited.

Reviewer: this line isn't necessary imo; if users want to change it they know why they want to change it. Either way, colon is being used after an incomplete clause

Author: I've removed it. Thanks for suggesting.

> Note that, if you're using cuttingedge Release Stream, if you change the `CustomFrameLimit` to any number above 999, it will be reverted to 999. On Stable, you can set the value above 1000 normally.
> Also, you can not switch into this setting with F7 shortcut in-game, even if you had done the steps above, (only available via the Settings menu). And you'll need to switch to Custom in the Settings before quitting the game, so it'll be available the next time osu! runs. Else, you'll have to change the `FrameSync` manually again.

Reviewer: ```suggestion
Note that, if you're using the Cutting Edge release stream, changing the `CustomFrameLimit` to any number above 999, will be reverted to 999; on Stable, you can set the value above 1000 with no issues.

Also note that you cannot switch into this setting with the `F7` shortcut in-game, even if you had done the steps above: you'll need to switch to Custom in the Settings before quitting the game so it'll be available the next time osu! runs. Otherwise, you'll have to change the `FrameSync` manually again.
```
Remark; also these should probably be hatnotes

Author: I've changed to hatnotes!
